src/test/java/io/cdap/plugin/gcp/spanner/connector/SpannerConnectorTest.java

```diff
@@ -64,35 +70,63 @@ public class SpannerConnectorTest {
   private static String table;
   private static String serviceAccountFilePath;
   private static String serviceAccountKey;
+  private static Spanner spanner;
+  private static TestEnvironment testEnvironment;
+  private static Schema expectedSchema;
+  private static Set<StructuredRecord> tableData;
 
   @BeforeClass
   public static void setupTestClass() throws Exception {
-    // Certain properties need to be configured otherwise the whole tests will be skipped.
-
-    String messageTemplate = "%s is not configured.";
-
-    project = System.getProperty("project.id");
-    if (project == null) {
-      project = System.getProperty("GOOGLE_CLOUD_PROJECT");
-    }
-    Assume.assumeFalse(String.format(messageTemplate, "project id"), project == null);
-
-    instance = System.getProperty("instance.id");
-    Assume.assumeFalse(String.format(messageTemplate, "instance id"), instance == null);
+    testEnvironment = TestEnvironment.load();
 
+    project = testEnvironment.getProject();
+    serviceAccountFilePath = testEnvironment.getServiceAccountFilePath();
+    serviceAccountKey = testEnvironment.getServiceAccountContent();
 
-    database = System.getProperty("database.name");
-    Assume.assumeFalse(String.format(messageTemplate, "database name"), database == null);
+    instance = "gcp-plugins-test-" + System.currentTimeMillis();
```

Comment:
same here - can we use UUID?

Reply:
changed to use timestamp in seconds followed by uuid. So it's still sorted, but also unique.